### ISSUE TITLE
Calendar invitation reply result not visually updated from attendee

### ISSUE THREAD

**Issue by Arsnael** [linagora/tmail-backend#1030]
### Related method call
`Calendar/reply`

### Environment
Production - tmail.linagora.com

### Reproducible steps
- Person A (any email provider) sends an event invitation to person B (linagora mail)
- Person B clicks any of the reply (yes/no/maybe) button from the email invitation
- Person B confirms the success of person A's reply by checking their email
- Both A & B check their calendar
    - B will see A's attend status
    - A will not see his/her own attend status

### Evidents
- [Reply video](https://drive.google.com/file/d/1U-HlgtsRJgb219qkV3AZs9-xu2z_SdRO/view?usp=sharing)
- Inviter success confirmation & calendar visualization:

![Image](https://github.com/linagora/tmail-backend/assets/160106668/873bbee0-ec98-4d01-8c96-074e9537334f)
![Image](https://github.com/linagora/tmail-backend/assets/160106668/1db33804-6f16-4021-bb63-0ab99c1361fd)
![Image](https://github.com/linagora/tmail-backend/assets/160106668/819ebc3a-08ed-4ff6-abdb-141de5f30a49)

- Invited (attendee) calendar visualization

![Image](https://github.com/linagora/tmail-backend/assets/160106668/d790162f-4da4-4890-8413-514ef0e6f66d)
![Image](https://github.com/linagora/tmail-backend/assets/160106668/c85f8ec6-1a80-4ea3-b22e-01cca743d01d)

**Comment by vttranlina** [comment]
can you update who is A, who is B in the Video, and the picture?
Sorry, A lot of pictures, but still confused about what you explain

**Comment by vttranlina** [comment]
Summary: A invites B, but A can not see A status on A calendar?

**Comment by Arsnael** [comment]
@vttranlina from what I understood from discussing it with dat a bit was:

```
Alice invite Bob 
Bob reply by Tmail blue Bar
Alice can see Bob reply in all cases (Alice is openpaas or non openpaas user)
But Bob can not see his status
```

After checking a bit I think, if you take the current behavior on prod:
- alice creates an envent and invites bob
- bob receives an email asking him his attendance to this event
- bob clicks on yes button in the blue bar
- opens a new tab with uri params telling openpaas calendar that bob attends this event, so openpaas cal does the update at that moment for bob (I think that's the current hack in place)
- sends a reply message with ics file on bob status to alice
- tmail process the mail for alice, detects the ics, extract the info, send it to rabbitmq, sabre consumes it and sync for alice about bob status

I think mobile team for testing is that when bob clicks on yes button, they use the jmap metthod for attendance, which replies an email with the ics that then tmail treats for alice and send to sabre to update bob attendance on alice account , but not opening a new tab with calendar uri with query param and not updating bob attendance status for bob on openpaas cal

Am I correct guys?

**Comment by vttranlina** [comment]
I got it, thank you.
I remember the purpose of the Jmap calendar reply was just to show the blue bar on Alice's side.
We didn't test the comunicate between tmail and openpass

**Comment by quantranhong1999** [comment]
> opens a new tab with uri params telling openpaas calendar that bob attends this event, so openpaas cal does the update at that moment for bob (I think that's the current hack in place)

I think it is not 100% related to the Calendar reply extension.

For example, I tested on prod (today is still using the hacky HTML extraction for the blue bar IMO):
- `quan.tranhong1999@gmail.com` (Google Calendar) invites `hqtran@linagora.com` (OpenPaas Calendar).
- `hqtran@linagora.com` click `Yes` on the blue bar to accept the invitation.
-  `quan.tranhong1999@gmail.com` (Google Calendar) updates the status of `hqtran@linagora.com` to Yes (attend).
- But `hqtran@linagora.com` (OpenPaas Calendar) does not update his status to Yes.

Can say this issue has been there the whole time and is not related to the Calendar reply extension IMO.

**Comment by chibenwa** [comment]
(Disclaimer: on holiday, I read the bug description however I did not read your comments)

That's expected!

We need to modify the ICS pipeline to also transmit the ICS generated by Bob to OpenPaaS. Sending the ICS as if it was received by Bob is likely enough. 

TLDR: we need a simple mailetcontainer modification for this.

If needed I will do it on monday.

**Comment by Arsnael** [comment]
I did think about it yesterday and tried to take a look on mailetcontainer.xml on prod in the cal section, but I wasn't completely sure. Might take an other look, if not support welcome monday

**Comment by Arsnael** [comment]
> We need to modify the ICS pipeline to also transmit the ICS generated by Bob to OpenPaaS. Sending the ICS as if it was received by Bob is likely enough.

Would be interested to know how... The only mail that gets processed is the one received by Alice from Bob. I really don't see how you send your ICS pretending it was received by Bob too. Not really keen on testing something I aint sure about on prod as well (no calendar on staging)

**Comment by chibenwa** [comment]
https://github.com/linagora/tmail-backend/pull/1033

Please team take care of the PR from now on now that dark magic is done...


https://github.com/linagora/tmail-backend/assets/6928740/17df1261-99e3-4026-a8ea-3a3003975778

**Comment by vttranlina** [comment]
I continue it on the existing Benoit PR https://github.com/linagora/tmail-backend/pull/1033
